### DISCUSSION TITLE
Temperature null when raw METAR has a defined temperature

### DISCUSSION THREAD

**Discussion by sdennstaedt** [weather-gov/api#369] [Q&A]
This call yields the json below...

https://api.weather.gov/stations/kclt/observations/latest?require_qc=true

Temperature is showing null although the temperature 5°C is defined in the METAR.
{
    "@context": [
        "https://raw.githubusercontent.com/geojson/geojson-ld/master/contexts/geojson-base.jsonld",
        {
            "wx": "https://api.weather.gov/ontology#",
            "s": "https://schema.org/",
            "geo": "http://www.opengis.net/ont/geosparql#",
            "unit": "http://codes.wmo.int/common/unit/",
            "@vocab": "https://api.weather.gov/ontology#",
            "geometry": {
                "@id": "s:GeoCoordinates",
                "@type": "geo:wktLiteral"
            },
            "city": "s:addressLocality",
            "state": "s:addressRegion",
            "distance": {
                "@id": "s:Distance",
                "@type": "s:QuantitativeValue"
            },
            "bearing": {
                "@type": "s:QuantitativeValue"
            },
            "value": {
                "@id": "s:value"
            },
            "unitCode": {
                "@id": "s:unitCode",
                "@type": "@id"
            },
            "forecastOffice": {
                "@type": "@id"
            },
            "forecastGridData": {
                "@type": "@id"
            },
            "publicZone": {
                "@type": "@id"
            },
            "county": {
                "@type": "@id"
            }
        }
    ],
    "id": "https://api.weather.gov/stations/KCLT/observations/2020-02-22T14:52:00+00:00",
    "type": "Feature",
    "geometry": {
        "type": "Point",
        "coordinates": [
            -80.950000000000003,
            35.219999999999999
        ]
    },
    "properties": {
        "@id": "https://api.weather.gov/stations/KCLT/observations/2020-02-22T14:52:00+00:00",
        "@type": "wx:ObservationStation",
        "elevation": {
            "value": 228,
            "unitCode": "unit:m"
        },
        "station": "https://api.weather.gov/stations/KCLT",
        "timestamp": "2020-02-22T14:52:00+00:00",
        "rawMessage": "KCLT 221452Z AUTO 31004KT 10SM CLR 05/M04 A3048 RMK AO2 SLP325 T00501039 50013",
        "textDescription": "Clear",
        "icon": "https://api.weather.gov/icons/land/day/skc?size=medium",
        "presentWeather": [],
        "temperature": {
            "value": null,
            "unitCode": "unit:degC",
            "qualityControl": "qc:Z"
        },
        "dewpoint": {
            "value": -3.8999999999999773,
            "unitCode": "unit:degC",
            "qualityControl": "qc:V"
        },
        "windDirection": {
            "value": 310,
            "unitCode": "unit:degree_(angle)",
            "qualityControl": "qc:V"
        },
        "windSpeed": {
            "value": 2.1000000000000001,
            "unitCode": "unit:m_s-1",
            "qualityControl": "qc:V"
        },
        "windGust": {
            "value": null,
            "unitCode": "unit:m_s-1",
            "qualityControl": "qc:Z"
        },
        "barometricPressure": {
            "value": 103220,
            "unitCode": "unit:Pa",
            "qualityControl": "qc:V"
        },
        "seaLevelPressure": {
            "value": 103250,
            "unitCode": "unit:Pa",
            "qualityControl": "qc:V"
        },
        "visibility": {
            "value": 16090,
            "unitCode": "unit:m",
            "qualityControl": "qc:C"
        },
        "maxTemperatureLast24Hours": {
            "value": null,
            "unitCode": "unit:degC",
            "qualityControl": null
        },
        "minTemperatureLast24Hours": {
            "value": null,
            "unitCode": "unit:degC",
            "qualityControl": null
        },
        "precipitationLastHour": {
            "value": null,
            "unitCode": "unit:m",
            "qualityControl": "qc:Z"
        },
        "precipitationLast3Hours": {
            "value": null,
            "unitCode": "unit:m",
            "qualityControl": "qc:Z"
        },
        "precipitationLast6Hours": {
            "value": null,
            "unitCode": "unit:m",
            "qualityControl": "qc:Z"
        },
        "relativeHumidity": {
            "value": null,
            "unitCode": "unit:percent",
            "qualityControl": "qc:C"
        },
        "windChill": {
            "value": null,
            "unitCode": "unit:degC",
            "qualityControl": "qc:V"
        },
        "heatIndex": {
            "value": null,
            "unitCode": "unit:degC",
            "qualityControl": "qc:V"
        },
        "cloudLayers": [
            {
                "base": {
                    "value": null,
                    "unitCode": "unit:m"
                },
                "amount": "CLR"
            }
        ]
    }
}

**Comment by StephenClouse** [comment]
We do not actually parse the METAR TAC code. We import this data from [MADIS](https://madis.ncep.noaa.gov/) in a structured format after it has gone through their processing and quality control.

What we have noticed is that we will sometimes receive the same METAR multiple times. The initial one will be missing certain data points that are obviously in the TAC, and later versions of the record will fill in these missing values. We are still talking to the MADIS team to see why this is occurring and whether this is a bug on their end.

Once we know more about why this is occurring and how (or if) we can mitigate it, we will adjust API or update our documentation appropriately.

The best thing I can recommend at the moment is to re-check the observation record periodically, as it may have been updated since first retrieval. Our next API release will support HTTP conditional requests on this endpoint via the Last-Modified header, so you can more easily poll the server to determine if there has been a change.

**Comment by zim2dive** [comment]
I was looking for a source of minimal "current weather" (temp, and maybe a short desc).  while I've had solid luck with the hourly forecasts from NWS, I did notice a lot of "null" values in the observations/latest... is this common?  (perhaps I'll try to find another source for the "now" values)
